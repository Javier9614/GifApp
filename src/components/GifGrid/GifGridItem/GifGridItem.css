.card{

    align-content: center;
    border: 1px solid grey;
    border-radius: 6px;
    overflow: hidden;
    margin-bottom: 10px;
    margin-right: 10px;
}
    
    .card h3{
        padding: 5px;
        text-align: center;
    }
    
    .card__img{
        display: flex;
        align-items: center;
    }

    .card__img img{
        height: 200px;
    }
    
    
