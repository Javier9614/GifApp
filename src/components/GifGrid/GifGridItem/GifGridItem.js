import React from 'react';
import PropTypes from 'prop-types';
import "./GifGridItem.css";
const GifGridItem = ({title,url}) => {
    
    return(
        <div className="card animate__animated animate__rotateInUpRight">
            <h3 className="animate__animated animate__rotateInUpRight">{title}</h3>
            <div className="card__img"> 
                <img src={url} alt={title}/>
            </div>
        </div>
    )
}
GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
export default GifGridItem;