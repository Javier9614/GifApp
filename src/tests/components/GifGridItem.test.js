import React from 'react';
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGrid/GifGridItem/GifGridItem";


describe('Puebas en <GifGridItem/>',() =>{

    const title = "Un titulo";
    const url = "https://localhost/gif.gif";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('Debe mostrar el componente <GifGridItem/>',()=>{


        expect(wrapper).toMatchSnapshot();

    })

    test('Debe tener un parrafo con el title',()=>{
        const p = wrapper.find('h3');
        expect( p.text().trim() ).toBe(title);

    })

    test('Debe tener una imagen con el url y el title',()=>{
        const img = wrapper.find('img');
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
        

    })

    test('Debe tener animate__rotateInUpRight',()=>{

        //.at es necesario si tuvieseis mas de un div como es mi caso
        //para elegir el div que quieres comprobar que tenga esa clase
        const div = wrapper.find('div').at(0);
        const h3 = div.find('h3');
        expect(div.hasClass(("animate__rotateInUpRight"))).toBe(true);
        expect(h3.hasClass(("animate__rotateInUpRight"))).toBe(true);


    })

})