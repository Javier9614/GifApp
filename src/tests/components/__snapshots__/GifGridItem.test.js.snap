// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <GifGridItem/> Debe mostrar el componente <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__rotateInUpRight"
>
  <h3
    className="animate__animated animate__rotateInUpRight"
  >
    Un titulo
  </h3>
  <div
    className="card__img"
  >
    <img
      alt="Un titulo"
      src="https://localhost/gif.gif"
    />
  </div>
</div>
`;
